pkg/crinit/util/util.go

```diff
@@ -170,6 +170,86 @@ func UpdateKubeconfig(pathOptions *clientcmd.PathOptions, name, endpoint,
 	return nil
 }
 
+// DeleteKubeconfigEntry helper to delete the kubeconfig file entry based on input
+// parameters.
+func DeleteKubeconfigEntry(out io.Writer, pathOptions *clientcmd.PathOptions, name,
+	kubeConfigPath string, dryRun, ignoreErrors bool) error {
+
+	pathOptions.LoadingRules.ExplicitPath = kubeConfigPath
+	kubeconfig, err := pathOptions.GetStartingConfig()
+	if err != nil {
+		return err
+	}
+
+	kubeconfigFile := pathOptions.GetDefaultFilename()
+	if pathOptions.IsExplicitFile() {
+		kubeconfigFile = pathOptions.GetExplicitFile()
+	}
+
+	if dryRun {
+		return nil
+	}
+
+	// If we are not going to ignore errors, then return an error immediately
+	// on the first error encountered. If ignoring errors, then only output
+	// errors when verbose logging is enabled. If it turns out that all three:
+	// context, cluster, and authinfo were not found, then return an error as
+	// there is no point in updating the kubeconfig and the ignore errors flag
+	// will also be checked by the caller.
+	errCount := 0
+
+	_, ok := kubeconfig.Contexts[name]
+	if !ok {
+		if !ignoreErrors {
+			return fmt.Errorf("cannot delete context %s, not in %s", name, kubeconfigFile)
+		}
+		glog.V(4).Infof("cannot delete context %s, not in %s", name, kubeconfigFile)
+		errCount++
+	} else {
+		delete(kubeconfig.Contexts, name)
+	}
+
+	_, ok = kubeconfig.Clusters[name]
+	if !ok {
+		if !ignoreErrors {
+			return fmt.Errorf("cannot delete cluster %s, not in %s", name, kubeconfigFile)
+		}
+		glog.V(4).Infof("cannot delete cluster %s, not in %s", name, kubeconfigFile)
+		errCount++
+	} else {
+		delete(kubeconfig.Clusters, name)
+	}
+
+	_, ok = kubeconfig.AuthInfos[name]
+	if !ok {
+		if !ignoreErrors {
+			return fmt.Errorf("cannot delete authinfo %s, not in %s", name, kubeconfigFile)
+		}
+		glog.V(4).Infof("cannot delete authinfo %s, not in %s", name, kubeconfigFile)
+		errCount++
+	} else {
+		delete(kubeconfig.AuthInfos, name)
+	}
+
+	if errCount == 3 {
+		return fmt.Errorf("cannot find %s context, cluster, and authinfo", name)
```

Comment:
Maybe update this error message to make it a bit more user friendly? Something like "Could not find any cluster registry context information (context, cluster or authinfo) in your kubeconfig."?

Reply:
I like it - done.